pages/page.py

```diff
@@ -98,15 +98,29 @@ def ajax_complete(self, driver):
 
         return not jquery_active
 
-    def click(self, element, scroll=True):
+    def click(self, element, scroll=True, click_attempts_max=2):
         """
         Click on the target element. If scroll is True, this will attempt to
-        scroll the element into view then click on it
-        """
-        if scroll:
-            self.scroll_to_element(element)
+         scroll the element into view then click on it.
 
-        element.click()
+        If the click attempt fails with WebDriverException then retry
+         scroll/click until the total number of attempts equals the max number
+         of attempts
+        """
+        attempts = 0
+        while attempts < click_attempts_max:
+            attempts += 1
+            if scroll:
+                self.scroll_to_element(element)
+
+            try:
+                element.click()
+                break
+            except WebDriverException:
+                # Raise the click exception if scrolling disabled or
+                # max number of scroll->click attempts
+                if not scroll or attempts >= click_attempts_max:
+                    raise
```

Comment:
Only thing I might do is create a custom exception so you say how many attempts were made, and maybe if there is some information on the element put that too.